include/pistache/common.h

```diff
@@ -37,7 +37,7 @@
 #define TRY_RET(...) \
     [&]() { \
         auto ret = __VA_ARGS__; \
-        if (ret < 0) { \
+        if ((ssize_t)ret < 0) { \
```

Comment:
Please use "static_cast<ssize_t>(ret)" instead of the older C-style casting.  Old-style casts are flagged as warnings when compiled with "-Weffc++".

Reply:
I just tried removing the cast all together and I don't see any warnings with neither g++ or clang - I guess that's a better fix.